### PR TITLE
Fix wrong asset 404 error

### PR DESCRIPTION
**Pull request by fflorent**
Reopen #176 with a fix to the issue @alexAubin mentioned here: https://github.com/YunoHost/package_check/pull/176#issuecomment-2429601078

### CODE REVIEW

File: lib/curl_tests.py [hunk 1]
```diff
@@ -229,18 +233,19 @@ def test(
             elif asset.startswith(f"{domain}/"):
                 asset = asset.replace(f"{domain}/", "")
             if not asset.startswith("/"):
-                asset = "/" + asset
+                asset = urljoin(base + "/", asset)
+            resolved_asset_url = f"https://{domain}{asset}"
```
Comment (fflorent) [inline lib/curl_tests.py:237]: What may work is using `urljoin` specifically here:
```suggestion
            resolved_asset_url = urljoin("https://{domain}", asset)
```

Wdyt?

Comment (fflorent) [reply lib/curl_tests.py:237]: To be honest, I don't remember why I committed that. If I am not wrong, `domain` should only contain a domain as its name suggest. So this should be safe.

Comment (Salamandar) [reply lib/curl_tests.py:237]: Indeed it should be safe. Lezdoit.

**Comment by Salamandar** [comment]
@alexAubin it should be good imho but if you want to check it out before merging.

File: lib/curl_tests.py (hunk 1 above)
Comment (fflorent) [reply lib/curl_tests.py]: I checked in run against my package, [Grist](https://github.com/YunoHost-Apps/grist_ynh), it seems to work (it passed the Installation steps, but not the upgrades or backup/restore, probably due to instabilities when running locally).

**Review by alexAubin — APPROVED** [review]
LGTM naively but we should keep a close eye on the CI after merging

**Comment by fflorent** [comment]
@alexAubin @Salamandar So the plan is that we merge this PR and in case of regression we revert? I am fine with that.

FWIW: I don't have the rights to merge, I can't do that myself.

**Comment by Salamandar** [comment]
Yeah, i'm just fixing some stuff that is already breaking the CI, then once i'm sure it works, let's merge this.
I don't want to have 2 issues to fix at the same time (if this PR is problematic) ;)

**Comment by fflorent** [comment]
Sounds wise, indeed!

**Comment by Salamandar** [comment]
looks like it passes :) https://ci-apps.yunohost.org/ci/job/11118

**Pull request merged**